frontend/src/pages/inventory/shared/Recipe.tsx

```diff
@@ -0,0 +1,73 @@
+import { useState } from 'react';
+import { Menu, MenuItem, Chip } from '@material-ui/core';
+import { v4 as uuid } from 'uuid';
+
+import './Recipe.scss';
+
+interface Ingredient {
+  id: number,
+  uuid: string,
+  name: string
+}
+
+export default function Recipe() {
+  // Replace with backend logic
+  const materials = [
+    { name: "Odyssey 1999 Brake", id: 1 },
+    { name: "Izumi ECO Chain 1/8\" Black", id: 2 },
+    { name: 'EAI Deluxe Steel Cog 1/8"', id: 3 },
+    { name: '2019 SL 1.1 Carbon Road Frame', id: 4 },
+    { name: 'All-City Gonzo Perforated Leather Saddle', id: 5 }
+  ];
+
+  const [recipe, setRecipe] = useState<Ingredient[]>([]);
+
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+  const handleClick = (event: React.MouseEvent<HTMLDivElement | HTMLLabelElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const addIngredient = (material: any) => {
+    const { name, id } = material;
+    setRecipe([...recipe, { name: name, id: id, uuid: uuid() }]);
+    handleClose();
+  }
+
+  const removeIngredient = (uuid: string) => {
+    setRecipe(recipe.filter(ingredient => ingredient.uuid !== uuid));
+  }
+
+  return (
+    <div className="Recipe">
+      <label onClick={handleClick}>Recipe</label>
+      <div role="button" className="recipe-input" tabIndex={0} onClick={handleClick}>
+        {recipe.map(ingredient =>
+          <Chip key={ingredient.uuid} label={ingredient.name} onDelete={() => removeIngredient(ingredient.uuid)} />
```

Comment:
Very Nice! One question, is there a reason why you used a uuid instead of simply using the item id?

Reply:
A recipe can use the same raw material more than once. For example, a bike can use a wheel twice, one in the front and one in the back.

Using the product Id is no longer appropriate in this case, since the two wheels would share Ids, and React would have no way to tell which chip in the DOM is associated with which wheel. A uuid was used to distinguish each individual wheel in the DOM and in the code.